ast/parser/EarthParser.g4

```diff
@@ -39,8 +39,34 @@ commandStmt:
 	| stopsignalStmt
 	| onbuildStmt
 	| healthcheckStmt
-	| shellStmt
-	| genericCommandStmt;
+	| shellStmt;
+
+// withStmt -------------------------------------------------------------------
+
+withStmt: withExpr (NL+ WS? withBlock)? NL+ WS? END;
+withBlock: stmts;
+
+withExpr: WITH WS withCommand;
+withCommand:
+	dockerCommand;
+
+dockerCommand: DOCKER (WS stmtWords)?;
+
+// ifStmt ---------------------------------------------------------------------
+
+ifStmt: ifClause (NL+ WS? elseIfClause)* (NL+ WS? elseClause)? NL+ WS? END;
+
+ifClause: IF WS ifExpr (NL+ WS? ifBlock)?;
```

Comment:
I'm not sure if i'm reading this correctly, but does this mean an ifBlock is optional? Does this mean one can do this:

```
IF test "$a" = "$b"
END
```
without any RUN or copy or any other statements in the code block?

Reply:
The nice thing about this, is if someone wanted to comment out some lines one could do:

```
IF test "$a" = "$b"
#    RUN something a user wants to temporarily comment out during development
END
```

Not sure exactly how I feel about this if that's the intended case; seems like it could work.

Reply:
Yes, exactly. I was wondering if we should allow this - I decided to allow given that all programming languages allow you to do this.